enclave-macro/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+extern crate proc_macro;
+	
+use proc_macro::TokenStream;
+
+#[proc_macro]
+pub fn get_network_id(_input: TokenStream) -> TokenStream {
```

Comment:
Any specific reason for using `proc-macro` instead of `macro_rules!`?

Reply:
because it takes an environment variable / string during compilation (say "aa") and then pastes it inside the source code as a u8 constant ("0xaa")